keps/sig-apps/0032-portable-service-definitions.md

```diff
@@ -0,0 +1,157 @@
+2. Secrets with a specific type and schema can be used to handle credentials and other relevant information for services that have them (e.g., a database). Not all services will require a secret (e.g., DNS)
+
+This subproject will list and document the resources and how controllers can implement them. This provides for interoperability including that for controllers and other tools, like validators, and a canonical listing held by a vendor neutral party.
+
+In addition to the resources, this subproject will also provide a controller implementing the defined services to support testing, providing an example implementation, and to support other Kubernetes subprojects (e.g., Minikube). Controller produced by this project are _not_ meant to be used in production.
+
+3rd party controllers implementing the CRDs and secrets can use a variety of methods to implement the service handling. This is where the Kubernetes Service Catalog can be an option. This subproject will not host or support 3rd party controllers but will list them to aide in users discovering them. This is in support of the 3rd user story.
+
+To support custom settings for services by a service provider and to add the ability to add default settings (user story 4) we are considering a pattern of using CRDs for a controller with configuration on a cluster wide and namespace based level. An example of this in existence today is the cert manager issuer and cluster issuer resources. How to support this pattern will be worked out as part of the process in the next step of building a working system. This pattern is tentative until worked out in practice.
+
+The next step is to work out the details on a common service and an initial process by which future services can be added. To work out the details we will start with MySQL and go through the process to make it work as managed by Kubernetes and each of the top three public clouds as defined by adoption. Public clouds and other cloud platforms following the top three are welcome to be involved in the process but are not required in the process.
+
+Before this KEP can move to being implementable at least 2 services need to go through the process of being implemented to prove out the process elements of this system.
+
+### Risks and Mitigations
+
+Two known risks include:
+
+1. The access controls and relationships between accounts and services. How will proper user and tenancy information be passed to clouds that require this form of information?
+2. Details required, when requesting a service, can vary between cloud providers that will implement this as a SaaS. How can that information be commonized or otherwise handled?
+
+## Graduation Criteria
+
+The following are the graduation criteria:
+
+- 5 organizations have adopted using the portable service definitions
+- Service definitions for at least 3 services have been created and are in use
```

Comment:
How can we measure "in use"?

Reply:
We could ask people. That's one way. The CNCF asks for a list of orgs using something but doens't require a method to report on it for CNCF projects.

Reply:
Thanks for the clarification, does it make sense to mention that in this KEP